Source/WebCore/css/parser/MutableCSSSelector.h

```diff
@@ -112,12 +112,13 @@ inline bool MutableCSSSelector::needsImplicitShadowCombinatorForMatching() const
 {
     return match() == CSSSelector::Match::PseudoElement
         && (pseudoElement() == CSSSelector::PseudoElement::UserAgentPart
+            || pseudoElement() == CSSSelector::PseudoElement::UserAgentPartLegacyAlias
+            || pseudoElement() == CSSSelector::PseudoElement::WebKitUnknown
```

Comment:
I would just remove this case. We don't need to look into the shadow DOM for webkit unknown pseudos.

Reply:
I still think we should remove this case, this is used to prepend implicit selectors to unknown ::-webkit pseudos, ... which shouldn't matter anyway if they don't match.